src/main/java/io/candydoc/ddd/SubdomainType.java

```diff
@@ -0,0 +1,7 @@
+package io.candydoc.ddd;
+
+public enum SubdomainType {
```

Comment:
A subdomain can be seen as a domain. It only indicates that this domain exists within a bigger set.

I dont thing there is a need to have a full `@Subdomaintype` annotation.

It might only be an optional property of the `@Subdomain` annotation.

Having a `@SubdomainType` without `@Subdomain` on a class does not make much sense.